### PR TITLE
Deploy script form frontend

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
image

image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,8 +47,18 @@
...
   "devDependencies": {
+    "@nomicfoundation/hardhat-chai-matchers": "^1.0.5",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.7",
+    "@nomicfoundation/hardhat-toolbox": "^2.0.1",
+    "@nomiclabs/hardhat-ethers": "^2.2.2",
+    "@nomiclabs/hardhat-etherscan": "^3.1.5",
+    "@typechain/ethers-v5": "^10.1.0",
+    "@typechain/hardhat": "^6.1.5",
+    "@types/aria-query": "^5.0.1",
+    "@types/mocha": ">=9.1.0",
```
Comment (TBomberman) [inline package.json:57]: Uneeded packages:
mocha, 
aria-query,
ethers-v5,
@typechain/hardhat,
hardhat-etherscan,
hardhat-network-helpers,
hardhat-chai-matchers
let's reduce the dependencies bloat

File: cache/solidity-files-cache.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (TBomberman) [inline cache/solidity-files-cache.json]: Do we need thsi file?

File: hardhat.config.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// testing the frontend.
+import "./tasks/accounts.ts";
+
+const config: HardhatUserConfig = {
+  solidity: "0.8.17",
+  networks: {
+    hardhat: {
+      chainId: 31337, // We set 1337 to make interacting with MetaMask simpler
+      accounts: {
+        mnemonic: "test test test test test test test test test test test junk",
```
Comment (TBomberman) [inline hardhat.config.ts]: Let's change the mnemonic to a different one from gs.

File: hardhat.config.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// The next line is part of the sample project, you don't need it in your
+// project. It imports a Hardhat task definition, that can be used for
+// testing the frontend.
+import "./tasks/accounts.ts";
+
+const config: HardhatUserConfig = {
+  solidity: "0.8.17",
+  networks: {
+    hardhat: {
+      chainId: 31337, // We set 1337 to make interacting with MetaMask simpler
```
Comment (TBomberman) [inline hardhat.config.ts]: chain id to 326, remove comment

File: hardhat.config.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      accounts: {
+        mnemonic: "test test test test test test test test test test test junk",
+        path: "m/44'/60'/0'/0",
+        initialIndex: 0,
```
Comment (TBomberman) [inline hardhat.config.ts]: can lines 18-29 be removed?

File: hardhat.config.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        count: 20,
+        passphrase: "",
+      },
+    },
+    GSChain: {
```
Comment (TBomberman) [inline hardhat.config.ts]: Remove GSChain to DSChain
chainId to 326
mnemonic to something else

File: package.json [hunk 7]
```diff
@@ -56,7 +68,12 @@
     "eslint-plugin-promise": "^6.1.1",
     "eslint-plugin-react": "^7.31.11",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "hardhat": "^2.12.6",
+    "hardhat-contract-sizer": "^2.7.0",
```
Comment (TBomberman) [inline package.json]: remove 
hardhat-contract-sizer, 
hardhat-gas-reporter,
solidity-coverage,

File: src/components/Stages/index.tsx [hunk 8]
```diff
@@ -138,7 +138,7 @@ function Stage() {
               <div className={style.stageData}>
                 <div className={style.stageTotalMoney}>
                   {stage?.stageContract.totalCommitted} ETH raised of{" "}
-                  {stage?.stage.stageGoal} ETH goal
+                  {stage?.stage?.stageGoal} ETH goal
```
Comment (TBomberman) [inline src/components/Stages/index.tsx]: Why do we have stage within a stage? Also the naming stageGoal is redundant if we already know the context is a stage.

File: src/utils/ipfsUtils.ts [hunk 9]
```diff
@@ -106,3 +76,21 @@ export const getStageData = async (
 
   return data;
 };
+
+const getIpfsClient = async (): Promise<IPFSHTTPClient> => {
+  const auth =
```
Comment (TBomberman) [inline src/utils/ipfsUtils.ts:81]: This seems like it is being called many times unecessarily.
Can it be cached and reused?

File: tasks/accounts.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { task } from "hardhat/config";
+
+task("accounts", "Prints the list of accounts", async (taskArgs, hre) => {
```
Comment (TBomberman) [inline tasks/accounts.ts]: how is this called?

File: package.json (hunk 1 above)
Comment (UsamaSaleem14) [reply package.json:57]: If I remove these dependencies I am getting the following error. So, these dependencies are being used under the hood.

Error HH801: Plugin @nomicfoundation/hardhat-toolbox requires the following dependencies to be installed: @nomicfoundation/hardhat-network-helpers, @nomicfoundation/hardhat-chai-matchers, @nomiclabs/hardhat-etherscan, @types/chai, @types/mocha, @typechain/ethers-v5, @typechain/hardhat.

File: cache/solidity-files-cache.json (hunk 2 above)
Comment (UsamaSaleem14) [reply cache/solidity-files-cache.json]: This file was created by the sh scripts when I ran to deploy the contract. We can add it in gitignore as well. Let me know about it

File: hardhat.config.ts (hunk 3 above)
Comment (UsamaSaleem14) [reply hardhat.config.ts:16]: Done

File: hardhat.config.ts (hunk 4 above)
Comment (UsamaSaleem14) [reply hardhat.config.ts]: Done

File: hardhat.config.ts (hunk 5 above)
Comment (UsamaSaleem14) [reply hardhat.config.ts]: Done

File: hardhat.config.ts (hunk 6 above)
Comment (UsamaSaleem14) [reply hardhat.config.ts]: Should I remove it or keep it? In the previous comment you suggested that we should remove from line 18-29

File: package.json (hunk 7 above)
Comment (UsamaSaleem14) [reply package.json]: Getting the following error upon removing these:

Error HH801: Plugin @nomicfoundation/hardhat-toolbox requires the following dependencies to be installed: hardhat-gas-reporter, solidity-coverage.

File: src/components/Stages/index.tsx (hunk 8 above)
Comment (UsamaSaleem14) [reply src/components/Stages/index.tsx]: Renamed it to stageData. The first stage is the name of the useState in the hook and the second stage is the name of the object inside of that useState hook. Changed stageGoal to goal

File: tasks/accounts.ts (hunk 10 above)
Comment (UsamaSaleem14) [reply tasks/accounts.ts]: We do not need it. Removed it

File: package.json (hunk 1 above)
Comment (TBomberman) [reply package.json:57]: ok, sorry, then keep whatever is needed.

File: hardhat.config.ts (hunk 3 above)
Comment (UsamaSaleem14) [reply hardhat.config.ts]: Should I use a specific mnemonic or whatever we want?

File: src/utils/ipfsUtils.ts (hunk 9 above)
Comment (UsamaSaleem14) [reply src/utils/ipfsUtils.ts:81]: Fixed

File: cache/solidity-files-cache.json (hunk 2 above)
Comment (TBomberman) [reply cache/solidity-files-cache.json]: Why is it not created in core when deploying contracts?

File: hardhat.config.ts (hunk 4 above)
Comment (TBomberman) [reply hardhat.config.ts]: Please also change for hardhat network.

File: hardhat.config.ts (hunk 3 above)
Comment (TBomberman) [reply hardhat.config.ts]: use "cat cat cat ... cat cat"

File: hardhat.config.ts (hunk 5 above)
Comment (TBomberman) [reply hardhat.config.ts]: image
Why is it still not removed?

File: hardhat.config.ts (hunk 6 above)
Comment (TBomberman) [reply hardhat.config.ts]: change it to "dog dog dog ... dog dog"

File: package.json (hunk 7 above)
Comment (TBomberman) [reply package.json]: Keep what we need.

File: src/context/IpfsContext.tsx [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import { IPFSHTTPClient } from "ipfs-http-client";
+import React from "react";
+
+const IpfsContext = React.createContext<IPFSHTTPClient | null>(null);
```
Comment (TBomberman) [inline src/context/IpfsContext.tsx]: Why are we still having this file now? Are we using the global cached client or we are using a client context?

File: hardhat.config.ts [hunk 12]
```diff
@@ -2,16 +2,11 @@ import { HardhatUserConfig } from "hardhat/config";
...
-// project. It imports a Hardhat task definition, that can be used for
-// testing the frontend.
-import "./tasks/accounts.ts";
-
 const config: HardhatUserConfig = {
   solidity: "0.8.17",
   networks: {
     hardhat: {
-      chainId: 31337, // We set 1337 to make interacting with MetaMask simpler
+      chainId: 31337,
```
Comment (TBomberman) [inline hardhat.config.ts]: Change id to 326.

File: src/context/IpfsContext.tsx (hunk 11 above)
Comment (UsamaSaleem14) [reply src/context/IpfsContext.tsx]: Yes I created a context. But it is now used the way we want. See, when ever we create a campaign and stage we redirect to that page which reloads the application. So, it does reinitiate everything and we are only using it in two places.

File: cache/solidity-files-cache.json (hunk 2 above)
Comment (TBomberman) [reply cache/solidity-files-cache.json]: add to git ignore

File: hardhat.config.ts (hunk 6 above)
Comment (TBomberman) [reply hardhat.config.ts]: did you change to DSChain?

**Comment by TBomberman** [comment]
Please also fix conflicts and merge from main. ty.

File: src/context/IpfsContext.tsx (hunk 11 above)
Comment (TBomberman) [reply src/context/IpfsContext.tsx]: Need to get rid of this file and all references as it is no longer needed.

**Pull request merged**